### PR TITLE
refactor: events to structs

### PR DESCRIPTION
**Pull request by Roznovjak**
Fixes: #326

### CODE REVIEW

File: pallets/duster/src/lib.rs [hunk 1]
```diff
@@ -193,13 +193,16 @@ pub mod pallet {
 	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
 	pub enum Event<T: Config> {
 		/// Account dusted.
-		Dusted(T::AccountId, T::Balance),
+		Dusted {
+			account_id: T::AccountId,
```
Comment (lumir-mrkva) [inline pallets/duster/src/lib.rs:197]: account_id -> who

File: pallets/lbp/src/lib.rs [hunk 2]
```diff
@@ -303,39 +303,57 @@ pub mod pallet {
...
-			AssetId,
-			AssetId,
-			BalanceOf<T>,
-			BalanceOf<T>,
-			AssetId,
-			BalanceOf<T>,
-		),
+		/// Pool was created by the `CreatePool` origin.
+		PoolCreated {
+			pool_account_id: PoolId<T>,
```
Comment (lumir-mrkva) [inline pallets/lbp/src/lib.rs:308]: pool_account_id -> id
pool_data -> data

File: pallets/marketplace/src/lib.rs [hunk 3]
```diff
@@ -133,7 +133,12 @@ pub mod pallet {
 
 			Prices::<T>::mutate_exists(class_id, instance_id, |price| *price = new_price);
 
-			Self::deposit_event(Event::TokenPriceUpdated(sender, class_id, instance_id, new_price));
+			Self::deposit_event(Event::TokenPriceUpdated {
+				owner: sender,
```
Comment (lumir-mrkva) [inline pallets/marketplace/src/lib.rs:137]: who

File: pallets/marketplace/src/lib.rs [hunk 4]
```diff
@@ -178,7 +183,13 @@ pub mod pallet {
 
 			<T as pallet_nft::Config>::Currency::reserve_named(&RESERVE_ID, &sender, amount)?;
 
-			Self::deposit_event(Event::OfferPlaced(sender, class_id, instance_id, amount, expires));
+			Self::deposit_event(Event::OfferPlaced {
+				offerer: sender,
```
Comment (lumir-mrkva) [inline pallets/marketplace/src/lib.rs:187]: who

File: pallets/marketplace/src/lib.rs [hunk 5]
```diff
@@ -178,7 +183,13 @@ pub mod pallet {
...
+				class_id,
+				instance_id,
+				price: amount,
```
Comment (lumir-mrkva) [inline pallets/marketplace/src/lib.rs:190]: price -> amount

File: pallets/marketplace/src/lib.rs [hunk 6]
```diff
@@ -307,32 +332,57 @@ pub mod pallet {
...
-		/// Offer was accepted \[sender, class_id, instance_id\]
-		OfferAccepted(T::AccountId, T::NftClassId, T::NftInstanceId, BalanceOf<T>),
-		/// Royalty hs been paid to the author \[class_id, instance_id, author, royalty, royalty_amount\]
-		RoyaltyPaid(T::NftClassId, T::NftInstanceId, T::AccountId, u8, BalanceOf<T>),
-		/// Marketplace data has been added \[class_type, sender, class_id, instance_id\]
-		RoyaltyAdded(T::NftClassId, T::NftInstanceId, T::AccountId, u8),
+		/// The price for a token was updated
+		TokenPriceUpdated {
+			owner: T::AccountId,
+			class_id: T::NftClassId,
```
Comment (lumir-mrkva) [inline pallets/marketplace/src/lib.rs:338]: class

File: pallets/marketplace/src/lib.rs [hunk 7]
```diff
@@ -307,32 +332,57 @@ pub mod pallet {
...
+			instance_id: T::NftInstanceId,
```
Comment (lumir-mrkva) [inline pallets/marketplace/src/lib.rs:339]: instance

File: pallets/xyk/src/lib.rs [hunk 8]
```diff
@@ -194,41 +194,65 @@ pub mod pallet {
...
+		},
+
+		/// Pool was created.
+		PoolCreated {
+			who: T::AccountId,
+			asset_a: AssetId,
+			asset_b: AssetId,
+			initial_shares_amount: Balance,
+			share_token: AssetId,
+			pool_account_id: T::AccountId,
```
Comment (lumir-mrkva) [inline pallets/xyk/src/lib.rs:221]: pool

**Review by lumir-mrkva — CHANGES_REQUESTED** [review]

**Review by enthusiastmartin — APPROVED** [review]
Let's sort out the versions and it is ready for merge .

**Pull request merged**